pkg/kubelet/cm/helpers_linux_test.go

```diff
@@ -205,50 +203,8 @@ func TestResourceConfigForPod(t *testing.T) {
 			},
 			enforceCPULimits: true,
 			quotaPeriod:      defaultQuotaPeriod,
-			expected:         &ResourceConfig{CpuShares: &guaranteedShares, CpuQuota: &guaranteedQuota, CpuPeriod: &defaultQuotaPeriod, Memory: &guaranteedMemory},
-		},
-		"guaranteed-no-cpu-enforcement": {
```

Comment:
I understand that with the change proposed in this PR we are no longer enforcing CPU limits on guaranteed pods and the test case `guaranteed-no-cpu-enforcement` is therefore a subset of when we just test for guaranteed pods. However, we could still keep this test case (`guaranteed-no-cpu-enforcement`) to make sure that (even thought it appears super obvious) we get the expected result (cpuQuota=-1) when both `enforceCPULimits= false` AND pod belongs to guaranteed QoS.